plugins/candela/web_client/views/CandelaWidget.js

```diff
@@ -0,0 +1,88 @@
+import View from 'girder/views/View';
+import { AccessType } from 'girder/constants';
+
+import CandelaParametersView from './CandelaParametersView';
+import CandelaWidgetTemplate from '../templates/candelaWidget.pug';
+import '../stylesheets/candelaWidget.styl';
+
+import candela from 'candela';
+import datalib from 'datalib';
+
+var CandelaWidget = View.extend({
+    events: {
+        'change .g-item-candela-component': 'updateComponent'
+    },
+
+    initialize: function (settings) {
+        this.item = settings.item;
+        this.accessLevel = settings.accessLevel;
+        this._components = [];
+        for (let component in candela.components) {
+            if (candela.components.hasOwnProperty(component)) {
+                this._components.push(component);
+            }
+        }
+
+        this.item.on('change', function () {
+            this.render();
+        }, this);
+
+        this.parametersView = new CandelaParametersView({
+            component: candela.components[this.$('.g-item-candela-component').val()],
+            parentView: this
+        });
+
+        this.render();
+    },
+
+    updateComponent: function () {
+        this.parametersView.setComponent(candela.components[this.$('.g-item-candela-component').val()]);
+    },
+
+    render: function () {
+        if (this.accessLevel >= AccessType.READ && this.item.get('name').toLowerCase().endsWith('.csv')) {
+            this.$el.html(CandelaWidgetTemplate({
+                components: this._components
+            }));
+            this.parametersView.setElement($('.g-item-candela-parameters'));
+            datalib.csv('/api/v1/item/' + this.item.get('_id') + '/download', (error, data) => {
```

Comment:
I think `this.item.downloadUrl()` would work here.  You might also consider adding an error handler.